csharp/src/Apache.Arrow/DecimalUtility.cs

```diff
@@ -0,0 +1,144 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Numerics;
+
+namespace Apache.Arrow
+{
+    /// <summary>
+    /// This is semi-optimised best attempt at converting to / from decimal and the buffers
+    /// </summary>
+    public static class DecimalUtility
+    {
+        private static readonly BigInteger _maxDecimal = new BigInteger(decimal.MaxValue);
+        private static readonly BigInteger _minDecimal = new BigInteger(decimal.MinValue);
+        private static readonly ulong[] s_powersOfTen =
+        {
+            1, 10, 100, 1000, 10000, 100000, 1000000, 10000000, 100000000, 1000000000, 10000000000, 100000000000,
+            1000000000000, 10000000000000, 100000000000000, 1000000000000000, 10000000000000000, 100000000000000000,
+            1000000000000000000, 10000000000000000000
+        };
+
+        private static int PowersOfTenLength => s_powersOfTen.Length - 1;
+
+        public static decimal GetDecimal(in ArrowBuffer valueBuffer, int index, int scale, int byteWidth,
+            bool isUnsigned = false)
```

Comment:
It looks like `isUnsigned` can be removed. No caller passes it in, and it doesn't appear to be used in this method.